main.go

```diff
@@ -68,20 +69,37 @@ func (m *MigrationRunner) Run(ctx context.Context, conn *sql.Conn, lastRanId int
 				continue
 			}
 			if _, err := conn.ExecContext(ctx, fileScanner.Text()); err != nil {
-				return fmt.Errorf("failed to execute migration %s: %w", file, err)
+				return updatedID, fmt.Errorf("failed to execute migration %s: %w", file, err)
 			}
 		}
 		log.Printf("Executed migration: %s\n", file)
+		updatedID++
```

Comment:
Could we use the id from the migration file name instead of an auto increment? That could save us from potential bugs down the line.